src/UsgsAstroProjectedLsSensorModel.cpp

```diff
@@ -0,0 +1,799 @@
+}
+
+//***************************************************************************
+// UsgsAstroProjectedLsSensorModel::imageToGround
+//***************************************************************************
+csm::EcefCoord UsgsAstroProjectedLsSensorModel::imageToGround(
+    const csm::ImageCoord& image_pt, double height, double desired_precision,
+    double* achieved_precision, csm::WarningList* warnings) const {
+  MESSAGE_LOG(
+      spdlog::level::info,
+      "Computing imageToGround for {}, {}, {}, with desired precision {}",
+      image_pt.line, image_pt.samp, height, desired_precision);
+
+  double x = 0, y = 0, z = 0;
+  double meterLine, meterSamp;
+  std::vector<double> meterCoord = pixelToMeter(image_pt.line, image_pt.samp, m_geoTransform);
+  meterLine = meterCoord[0];
+  meterSamp = meterCoord[1];
+  PJ_CONTEXT *C = proj_context_create();
+
+  /* Create a projection. */
+  PJ *isdProj = proj_create(C, (m_projString + " +type=crs").c_str());
+  if (0 == isdProj) {
+    MESSAGE_LOG(
+        spdlog::level::debug,
+        "Failed to create isd transformation object");
+    return csm::EcefCoord(x, y, z);
+  }
+
+  /* Create the geocentric projection for our target */
+  std::string radius_a = "+a=" + std::to_string(m_majorAxis);
+  std::string radius_b = "+b=" + std::to_string(m_minorAxis);
+  std::string projString = "+proj=geocent " + radius_a + " " + radius_b + " +type=crs";
+  PJ *ecefProj = proj_create(C, projString.c_str());
+  if (0 == ecefProj) {
+    MESSAGE_LOG(
+        spdlog::level::debug,
+        "Failed to create geocent transformation object");
+    return csm::EcefCoord(x, y, z);
+  }
+
+  // Compute the transformation from our ISIS projection to ecef
+  PJ *isdProj2ecefProj = proj_create_crs_to_crs_from_pj(C, isdProj, ecefProj, 0, 0);
+  PJ_COORD c_in;
+  c_in.xy.x = meterSamp;
+  c_in.xy.y = meterLine;
+  PJ_COORD c_out = proj_trans(isdProj2ecefProj, PJ_FWD, c_in);
+  x = c_out.xyz.x, y = c_out.xyz.y, z = c_out.xyz.z;
+  MESSAGE_LOG(
+      spdlog::level::info,
+      "imageToGround result {} {} {}",
+      x, y, z);
+  return csm::EcefCoord(x, y, z);
+}
+
+//***************************************************************************
+// UsgsAstroProjectedLsSensorModel::imageToGround
+//***************************************************************************
+csm::EcefCoordCovar UsgsAstroProjectedLsSensorModel::imageToGround(
+    const csm::ImageCoordCovar& image_pt, double height, double heightVariance,
+    double desired_precision, double* achieved_precision,
+    csm::WarningList* warnings) const {
+  csm::EcefCoord groundCoord = imageToGround(image_pt, height);
+  csm::ImageCoord cameraImagePt = UsgsAstroLsSensorModel::groundToImage(groundCoord);
+  csm::ImageCoordCovar cameraImagePtCovar(cameraImagePt.line, cameraImagePt.samp);
+  
+  return UsgsAstroLsSensorModel::imageToGround(cameraImagePtCovar, height, heightVariance, desired_precision, achieved_precision, warnings);
```

Comment:
How is performance when having to go back and forth between image and ground?

Reply:
No idea, fast enough that working with the camera in a jupyter notebook is not noticeable